src/main/java/com/conveyal/r5/analyst/scenario/RasterCost.java

```diff
@@ -0,0 +1,120 @@
+package com.conveyal.r5.analyst.scenario;
+
+import com.conveyal.analysis.components.WorkerComponents;
+import com.conveyal.file.FileStorageFormat;
+import com.conveyal.file.FileStorageKey;
+import com.conveyal.r5.rastercost.CostField;
+import com.conveyal.r5.rastercost.ElevationLoader;
+import com.conveyal.r5.rastercost.RasterDataSourceSampler;
+import com.conveyal.r5.rastercost.SunLoader;
+import com.conveyal.r5.streets.EdgeStore;
+import com.conveyal.r5.transit.TransportNetwork;
+import org.geotools.referencing.operation.builder.LocalizationGrid;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.util.ArrayList;
+
+import static com.conveyal.file.FileCategory.DATASOURCES;
+import static com.conveyal.r5.analyst.scenario.RasterCost.CostFunction.MINETTI;
+import static com.conveyal.r5.analyst.scenario.RasterCost.CostFunction.SUN;
+import static com.conveyal.r5.analyst.scenario.RasterCost.CostFunction.TOBLER;
+
+/**
+ * Custom (experimental) Modification that loads costs from a raster.
+ * As a custom modification, the model class only exists in R5, there is no separate UI/backend modification class.
+ */
+public class RasterCost extends Modification {
+
+    private static final Logger LOG = LoggerFactory.getLogger(RasterCost.class);
+
+    /**
+     * The ID of the DataSource, which must be a geotiff raster.
+     * This will define the field over which to evaluate the costs.
+     * It is assumed to be of type GEOTIFF because the workers don't have access to the DataStore metadata.
+     */
+    public String dataSourceId;
+
+    public enum CostFunction {
+        TOBLER, MINETTI, SUN
+    }
+
+    /**
+     * The function that transforms a series of values along a road segment to a cost that will be added onto the
+     * base street edge traversal cost.
+     */
+    public CostFunction costFunction;
+
+    /**
+     * Scale the raster values before passing them to the cost function. For example, the input raster file
+     * may be in decimeters to make it more compact with less artificial precision.
+     */
+    public double inputScale = 1;
+
+    /**
+     * Scale the cost after applying the function rather than the input to calibrate cost, uniformly increasing or
+     * decreasing it. Distinct from inputScale since cost functions are not necessarily linear.
+     */
+    public double outputScale = 1;
+
+    /** @see com.conveyal.r5.rastercost.RasterDataSourceSampler#setNorthShiftMeters(double) */
+    public double northShiftMeters = 0;
+
+    /** @see com.conveyal.r5.rastercost.RasterDataSourceSampler#setEastShiftMeters(double) */
+    public double eastShiftMeters = 0;
+
+    private CostField.Loader loader;
+
+    @Override
+    public boolean resolve (TransportNetwork network) {
+        if (costFunction == TOBLER || costFunction == MINETTI) {
+            loader = new ElevationLoader(dataSourceId);
```

Comment:
This means passing an enum value that has other non-elevation values (SUN) and unfortunately Java doesn't allow defining enums as unions of other enums (so CostFunction could be the union of ElevationCostFunction and SUN). We do want to make this configurable, so I passed in an instance of the function itself rather than the CostFunction enum. This has future thread safety implications, but I documented those.